contracts/messaging/gateway/L1ERC20Gateway.sol

```diff
@@ -21,18 +29,23 @@ contract L1ERC20Gateway is FuelMessagesEnabled, Ownable, Pausable {
     /// @notice Maps ERC20 tokens to Fuel tokens to balance of the ERC20 tokens deposited
     mapping(address => mapping(bytes32 => uint256)) public s_deposits;
 
-    /////////////////
-    // Constructor //
-    /////////////////
+    /////////////////////////////
+    // Constructor/Initializer //
+    /////////////////////////////
 
-    /// @notice Contract constructor to setup immutable values
+    /// @notice Constructor disables initialization for the implementation contract
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
```

Comment:
I forget the exact reason this is recommended, but it's what openzepplins contract wizard currently auto generates. There may be multiple reasons, but one I remember reading is to prevent confusion as to what is just an implementation for a proxy with the idea being that functions will error is the state has not been initialized.